includes/CLI/Plugin_Check_Command.php

```diff
@@ -0,0 +1,339 @@
+	 *
+	 * [--ignore-warnings]
+	 * : Limit displayed results to exclude warnings.
+	 *
+	 * [--ignore-errors]
+	 * : Limit displayed results to exclude errors.
+	 *
+	 *
+	 * ## EXAMPLES
+	 *
+	 *   wp plugin check akismet
+	 *   wp plugin check akismet --check=escaping
+	 *   wp plugin check akismet --format=json
+	 *
+	 * @subcommand check
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param array $args List of the positional arguments.
+	 * @param array $assoc_args List of the associative arguments.
+	 *
+	 * @throws Exception Throws exception.
+	 */
+	public function check( $args, $assoc_args ) {
+
+		$assoc_args = $this->get_options( $assoc_args );
+
+		Plugin_Request_Utility::get_plugin_basename_from_input( $args[0] );
+		Plugin_Request_Utility::initialize_runner();
+
+		$cli_runner = Plugin_Request_Utility::get_runner();
+
+		try {
+
+			$result = $cli_runner->run();
+
+		} catch ( Exception $error ) {
+
+			WP_CLI::error( $error->getMessage() );
+		}
+
+		// Get errors and warnings from the results.
+		$errors = array();
+		if ( empty( $assoc_args['ignore-errors'] ) ) {
+			$errors = $result->get_errors();
+		}
+		$warnings = array();
+		if ( empty( $assoc_args['ignore-warnings'] ) ) {
+			$warnings = $result->get_warnings();
+		}
+
+		// Get formatter.
+		$formatter = $this->get_formatter( $assoc_args );
+
+		// Print the formatted results.
+		// Go over all files with errors first and print them, combined with any warnings in the same file.
+		foreach ( $errors as $file_name => $file_errors ) {
+			$file_warnings = array();
+			if ( isset( $warnings[ $file_name ] ) ) {
+				$file_warnings = $warnings[ $file_name ];
+				unset( $warnings[ $file_name ] );
+			}
+			$file_results = $this->flatten_file_results( $file_errors, $file_warnings );
+			$this->display_results( $formatter, $file_name, $file_results );
+		}
+
+		// If there are any files left with only warnings, print those next.
+		foreach ( $warnings as $file_name => $file_warnings ) {
+			$file_results = $this->flatten_file_results( array(), $file_warnings );
+			$this->display_results( $formatter, $file_name, $file_results );
+		}
+	}
+
+	/**
+	 * Validate associative arguments.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param array $assoc_args List of the associative arguments.
+	 * @return array List of the associative arguments.
+	 *
+	 * @throws WP_CLI\ExitException Show error if plugin not found.
+	 */
+	protected function get_options( $assoc_args ) {
+
+		$options = array(
+			'checks'          => 'all',
+			'flag'            => 'stable',
+			'format'          => 'table',
+			'ignore-warnings' => false,
+			'ignore-errors'   => false,
+		);
+		$options = wp_parse_args( $assoc_args, $options );
+
+		if ( ! in_array( $options['flag'], $this->check_flags, true ) ) {
+
+			WP_CLI::error(
+				sprintf(
+					// translators: 1. Check flags.
+					__( 'Invalid flag argument, valid value will be one of [%1$s]', 'plugin-check' ),
+					implode( ', ', $this->check_flags )
+				)
+			);
+		}
```

Comment:
@vishalkakadiya I see we have this option to choose what level of check stability we can run (stable, experimental etc) though we do not use this is the `check` method here.

I will discuss with @felixarntz on what needs to be done here to implement this or whether we even want to implement this as part of milestone 1.

Reply:
@jjgrainger Sure, we can discuss this with @felixarntz. Meanwhile, I have fixed all your other feedback. Thanks!